### DISCUSSION TITLE
Access point response

### DISCUSSION THREAD

**Discussion by mauroForlimpopoli** [Moddable-OpenSource/moddable#1107] [Ideas]
My Moddable Two is set as Server and this code works correctly.

```
    let myWeb = '<!DOCTYPE html>'+
    '<html>'+
    '<head>'+
    '<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>'+
    '<script>'+
    '$(document).ready(function(){'+
      '$("button").click(function(){'+
        '$("p").hide("slow", function(){'+
         ' alert("' + valoreWeb + '");'+
       '});'+
      '});'+
    '});'+
    '</script>'+
    '</head>'+
    '<body>'+
    
    '<button>Hide</button>'+
    
    '</body>'+
    '</html>'

      let serverAP = new Server({});
		serverAP.callback = function (message, value)
		{	
                      if (Server.prepareResponse === message) {
	                      return {headers: ["Content-type", "text/html"], body: myWeb };
                      }
               }
```
"valoreWeb" is a variable of my project.  **Very important** !!!

Questions:
1) How can I modify the code if the device is an Access Point ? (no internet connection)
The problem is how to include the file "jquery.min.js" in the memory of Moddable Two and the syntax to include it in the code
I think is possible to include it in the memory like a Resource, but how include it in the code ?
2) Can I add a background image of the body? (if saved in the Moddable memory)
Probably is the same problem of above

PS: Having  "valoreWeb" as a variable I can't create a page.html and use it

Thank you

**Comment by phoddie** [comment]
To be sure I understand, you want to have the resource `https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js` available when the user is not connected to the Internet? These examples should help:

- The [HTTP Server Resource example](https://github.com/Moddable-OpenSource/moddable/blob/public/examples/network/http/httpserverresource/main.js) shows how to return a large text file stored in a resource.
- The [HTTP ZIP example](https://github.com/Moddable-OpenSource/moddable/tree/public/examples/network/http/httpzip) shows how to server a web site from a ZIP file stored in a Resource. This is great because it allows the data to be compressed and supports client-side caching with ETag.

**Comment by mauroForlimpopoli** [comment]
Hi Peter, the examples are great but I have another issue.
Cause "valoreWeb " variable I can't use a page.html from memory. I need to create dynamically
I simplified the code.

    let myWeb = '<!DOCTYPE html>'+
    '<html>'+
    '<head>'+
    '</head>'+
    '<body background="logo.png">'   ***** "logo.png is a picture I want in background ... how to do ??
     +  valoreWeb  +                                    **** variable of project
    '</body>'+
    '</html>'

 let serverAP = new Server({});
		serverAP.callback = function (message, value)
		{	
                      if (Server.prepareResponse === message) {
	                      return {headers: ["Content-type", "text/html"], body: myWeb };
                      }
               }
I thing something like:
let pict= new Resource("logo.png");
and include it like....
```
'<body background=" ' + pict +  ' ">'  +
```
Thanks in advance

**Reply by phoddie** [reply]
> Cause "valoreWeb " variable I can't use a page.html from memory. I need to create dynamically

By "memory" I understand you to mean "flash memory" and not "RAM".

You can achieve the same result using the httpzip example by injecting the variables using WebSockets. It works really nicely. But perhaps client-side dynamic HTML it is more sophisticated than you want for this project.

Your logo.png **is** static data. There's no reason to load it into RAM to embed into the HTML (which would require base64 encoding, increasing the data size).  You should serve it from a resource referenced by a relative URL from your HTML. The first example above shows how to serve static data from resources.
